src/Feature/MarkdownHeader/MarkdownHeaderManager.ts

```diff
@@ -58,7 +58,18 @@ export class MarkdownHeaderManager extends AbstractManager {
         const promises = [];
         for (const view of views) {
             if (!path || view.file.path === path) {
-                promises.push((async () => this.resolver.resolve(view.file.path))().then(r => this.setTitle(view, r)));
+                promises.push(
+                    (async () => {
+                        const path = view.file.path;
+                        const title = await this.resolver.resolve(path);
+                        const currentPath = view.file?.path;
+                        if (currentPath === path) {
+                            this.setTitle(view, title);
```

Comment:
We can put `await this.resolver.resolve(path)` here to avoid wasting of resources in case path !== currentPath.

Reply:
@snezhig `currentPath` might change because view is closed or opened another file, that's why the bug occurred intermittently

From the other hand, I don't even understand why do we need promises here. All the operations here are synchronous. `await` just acts as `setImmediate`. I don't know if it was your intent

Reply:
I meant just move line 64 to 67.

The first feature was Explorer which has many items to update and we need to have async functions here. Another features work the same way.
I think that it is better to separate logic into update and init in the future to make the logic more flexible and understanding.